cpp/src/column/column_view.cpp

```diff
@@ -76,6 +77,24 @@ size_type column_view_base::null_count(size_type begin, size_type end) const
            ? 0
            : cudf::count_unset_bits(null_mask(), offset() + begin, offset() + end);
 }
+
+// simple prime number multiplication algorithm.
+// Adapted from http://myeyesareblind.com/2017/02/06/Combine-hash-values/#apachecommons
+constexpr void combine_hash(size_t& h1, size_t h2) { h1 = h1 * 127 + h2; }
```

Comment:
Would also be useful to link this in the impl of `struct hash<cudf::data_type>`. Also, how did you choose the multiplier? Is it an arbitrary prime number, or is it related to the number of bits required to encode the scale of a decimal in `type_id`?

Reply:
Yes. will do it. This common `combine_hash` is needed anyway to use across our library.

`hash = hash * odd_prime + c;`
as explained at 
https://www.google.co.in/books/edition/Effective_Java/piOyzYqeZGgC?hl=en&gbpv=1&pg=PA48&printsec=frontcover

127 is next prime with similar properties as 31. `127*i = i<<7-i`, so I choose 127. 
It's not related to any data type used here.

Similar fast alternatives are [boost hash_combine](https://www.boost.org/doc/libs/1_35_0/doc/html/boost/hash_combine_id241013.html), fnv ([link](http://myeyesareblind.com/2017/02/06/Combine-hash-values/#3-performance-on-artificial-test-data)). 
boost hash_combine is very commonly used (need to adapt for 64 bit).